src/Lib/Effects/Measure.hs

```diff
@@ -0,0 +1,34 @@
+{-# LANGUAGE BangPatterns     #-}
+{-# LANGUAGE FlexibleContexts #-}
+
+module Lib.Effects.Measure (timedAction) where
+
+import           System.CPUTime              (getCPUTime)
+
+import           Lib.App                     (AppEnv, ekgStore, timings)
+
+import qualified Data.HashMap.Strict         as HashMap
+import qualified System.Metrics              as Metrics
+import qualified System.Metrics.Distribution as Distribution
+
+-- | Performs action
+timedAction :: (MonadReader AppEnv m, MonadIO m) => Text -> m a -> m a
+timedAction label action = do
+  start <- liftIO getCPUTime
+  !result <- action
+  end <- liftIO getCPUTime
+  let !timeTaken = fromIntegral (end - start) * 1e-12
+  dist <- getOrCreateDistribution label
+  liftIO $ Distribution.add dist timeTaken
+  return result
```

Comment:
Could you please now use 4 spaces here in the body of this function? Now it's almost impossible to notice `where`